test-app/src/main/java/org/readium/r2/testapp/utils/LifecycleMediaSessionService.kt

```diff
@@ -0,0 +1,56 @@
+package org.readium.r2.testapp.utils
+
+import androidx.lifecycle.ServiceLifecycleDispatcher
+import androidx.media2.session.MediaSessionService
+import androidx.lifecycle.Lifecycle
+
+import androidx.annotation.CallSuper
+
+import android.content.Intent
+
+import android.os.IBinder
+import androidx.lifecycle.LifecycleOwner
+
+
+abstract class LifecycleMediaSessionService : MediaSessionService(), LifecycleOwner {
+
+    @Suppress("LeakingThis")
+    private val lifecycleDispatcher = ServiceLifecycleDispatcher(this)
+
+    @CallSuper
+    override fun onCreate() {
+        lifecycleDispatcher.onServicePreSuperOnCreate()
+        super.onCreate()
+    }
+
+    @CallSuper
+    override fun onBind(intent: Intent): IBinder? {
+        lifecycleDispatcher.onServicePreSuperOnBind()
+        return super.onBind(intent)
+    }
+
+    @CallSuper
+    override fun onStart(intent: Intent?, startId: Int) {
+        lifecycleDispatcher.onServicePreSuperOnStart()
+        super.onStart(intent, startId)
+    }
+
+    // this method is added only to annotate it with @CallSuper.
+    // In usual service super.onStartCommand is no-op, but in LifecycleService
+    // it results in mDispatcher.onServicePreSuperOnStart() call, because
```

Comment:
But I don't think `onStart` is ever called in this case, thus `lifecycleDispatcher.onServicePreSuperOnStart()` is never called above.

Reply:
Why do you think `onStart` is never called?
I mostly copy-pasted this code from `LifeycleService` (and should have written so):
https://android.googlesource.com/platform/frameworks/support/+/refs/heads/androidx-main/lifecycle/lifecycle-service/src/main/java/androidx/lifecycle/LifecycleService.java